File: web/static/js/components/site_view.js
import React  from "react";

export default class SiteView extends React.Component {
  render() {
    return(
      <div>
        <h2>{this.props.site.protocol + "://" + this.props.site.domain}</h2>
      </div>
    );
  }
}

SiteView.propTypes = {
  site: React.PropTypes.object.isRequired
};

File: web/views/api/v1/site_view.ex
defmodule PhxOembed.Api.SiteView do
  use PhxOembed.Web, :view

  def render("show.json", %{site: site}) do
    %{
      id: site.id,
      domain: site.domain,
      protocol: site.protocol
    }
  end

  def render("index.json", %{sites: sites}) do
    render_many(sites, PhxOembed.Api.SiteView, "show.json")
  end

  def render("error.json", %{error: error}) do
    %{error: error}
  end
end
